### ISSUE TITLE
AvoidDefaultSerializableInInnerClasses false alarm for classes in interfaces

### ISSUE THREAD

**Issue by PhilippWendler** [sevntu-checkstyle/sevntu.checkstyle#1119]
The check AvoidDefaultSerializableInInnerClasses is supposed to match only (non-static) inner classes, but in [its code](https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/96499a1feb6a1d6360f32e8636cf692c9d84545e/sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/AvoidDefaultSerializableInInnerClassesCheck.java#L97-L112) only checks for explicit `static` modifiers. So it mistakenly matches on static nested classes that are implicitly static and do not have an explicit `static` modifier, such as classes in interfaces (cf. [JLS 8.1.3](https://docs.oracle.com/javase/specs/jls/se17/html/jls-8.html#jls-8.1.3)).

Example:
```java
import java.io.Serializable;

public interface Test {

  class TestClass implements Serializable {}
}
```
No warning should be raised here, but version 1.44.1 does raise a warning: "Inner class should not implement default Serializable interface".

**Comment by romani** [comment]
Please follow issue template to show all by terminal and CLI output

**Comment by romani** [comment]
fix is merged.
